tools/test/audio/comp_run.sh

```diff
@@ -35,16 +108,30 @@ else
 fi
 
 # Run command
-# There is no more need to specify the library explicitly
-#LIB="-a ${COMP}=libsof_${COMP}.so"
-LIB=""
-ARG="-d -r $FS1 -R $FS2 -i $FN_IN -o $FN_OUT -t $TPLG $BINFMT $LIB"
+OPTS="$DEBUG -r $FS_IN -R $FS_OUT -c $CHANNELS_IN -n $CHANNELS_OUT $BINFMT -t $TPLG"
+DATA="-i $FN_IN -o $FN_OUT"
+ARG="$OPTS $EXTRA_OPTS $DATA"
 CMD="$HOST_EXE $ARG"
+VALGRIND_CMD="valgrind --leak-check=yes --error-exitcode=1"
 export LD_LIBRARY_PATH=$HOST_LIB:$TPLG_LIB
 
 # Run test bench
 echo "Command:         $HOST_EXE"
 echo "Argument:        $ARG"
-echo "LD_LIBRARY_PATH: ${LD_LIBRARY_PATH}"
+echo "LD_LIBRARY_PATH=${LD_LIBRARY_PATH}"
 
-valgrind --leak-check=yes --error-exitcode=1 $CMD
+rm -f "$FN_OUT"
+rm -f "$FN_TRACE"
+if [ "$VALGRIND" = "yes" ]; then
+    if [ -z "$FN_TRACE" ]; then
+	$VALGRIND_CMD $CMD
+    else
+	$VALGRIND_CMD $CMD 2> "$FN_TRACE"
```

Comment:
Not yet done, valgrind outputs into beginning and end, with experience it's easy to spot which is it. But obviously it's not the case for occasional users. The testbench output is quite a mess today, some trace and markup language prints appear mixed. I don't know who needs the markup. It would need a big cleanup with redirect of everything with simulated time stamps.